src/transformation/visitors/function.ts

```diff
@@ -196,7 +197,7 @@ export function transformFunctionToExpression(
         node
     );
     const functionExpression = lua.createFunctionExpression(
-        lua.createBlock(transformedBody),
+        lua.createBlock(isAsyncFunction(node) ? wrapInAsyncAwaiter(context, transformedBody) : transformedBody),
```

Comment:
Can you move this to an extra variable or fucntion?
It should be more clear that async is handled here, otherwise it's easy to over read.